pyfar/classes/audio.py

```diff
@@ -796,6 +811,45 @@ def sampling_rate(self):
     def sampling_rate(self, value):
         self._sampling_rate = value
 
+    @property
+    def complex(self):
+        return self._complex
+
+    @complex.setter
+    def complex(self, value):
+        # from complex=True to complex=False
+        if self._complex and not value:
+            if self._domain == 'time':
+                if np.all(np.abs(np.imag(self._data))) < 1e-15:
+                    self._complex = value
+                    self._data = self._data.astype(float)
+                else:
+                    raise ValueError("Signal has complex-valued time data"
+                                     "complex flag connot be `False`.")
+            if self._domain == 'freq':
+                # check for conjugate symmetry
+                if self._check_conjugate_symmetry():
+                    # and remove rendundant part of the spectrum
+                    self._data = fft._remove_mirror_spectrum(self._data)
```

Comment:
I think we need to add a test to check if data at 0 Hz and Nyquist are real valued (something like `abs(imag(data)) < 1e-15`)